proposals/lip-0045.md

```diff
@@ -984,164 +839,223 @@ The Interoperability module contains the following commands:
+        "fee": fee,
+        "status": status,
+        "params": encode(paramsSchema, params)
+    }
+
+    # Validate ccm size
+    validateFormat(ccm)
+
+    # Pay message fee
+    Token.payMessageFee(sendingAddress, ccm)
+
+    addToOutbox(partnerChainID, ccm)
+    ownChainAccount.nonce += 1
+
+    # Emit CCM processed event.
+    ccmID = SHA-256(encode(crossChainMessageSchema, ccm))
+
+    emitEvent(
+        moduleID = MODULE_ID_INTEROPERABILITY,
+        typeID = TYPE_ID_CCM_PROCESSED,
+        data = {"ccmID": ccmID, "status": CCM_PROCESSED_STATUS_SENT},
+        topics = [ccm.sendingChainID, ccm.receivingChainID, ccmID]
     )
 ```
 
 #### terminateChain
 
-For the function description, parameters, and return value see the corresponding internal function `terminateChainInternal`.
-
-##### Execution
+The `terminateChain` function terminates a chain account.
 
 ```python
-terminateChain(chainID):
+def terminateChain(chainID: ChainID) -> None:
+    # Chain was already terminated, do nothing
     if terminatedStateAccount(chainID) exists:
-        return False
+        return
 
     if chainID == ownChainAccount.ID:
-        return False
+        raise Exception('Chain cannot terminate itself.')
 
     # Processing on a sidechain
     if ownChainAccount.ID != MAINCHAIN_ID:
         # The mainchain cannot be terminated
         if chainID == MAINCHAIN_ID:
-            return False
+            raise Exception('The mainchain cannot be terminated.')
 
-    return terminateChainInternal(chainID)
-```
+    send(
+        EMPTY_FEE_ADDRESS,
+        MODULE_ID_INTEROPERABILITY,
+        CROSS_CHAIN_COMMAND_ID_CHANNEL_TERMINATED,
+        chainID,
+        0,
+        CCM_STATUS_OK,
+        EMPTY_BYTES
+    )
 
-#### getChainAccount
+    return createTerminatedStateAccount(chainID)
+```
 
-The `getChainAccount` function returns the object stored in the chain data substore with store key equal to the input `chainID`.
+#### error
 
-##### Parameters
+The `error` function is used to add an error code to a CCM and then add it to the outbox of a partner chain.
 
-* `chainID`: The ID of the chain.
+```python
+def error(ccm: CCM, errorStatus: uint32, relatedCCMIDs: list[bytes]) -> None:
+    # Error codes from 0 to MAX_RESERVED_ERROR_STATUS (included) are reserved to the Interoperability module
+    if 0 <= errorStatus <= MAX_RESERVED_ERROR_STATUS:
+        raise Exception('Invalid error status.')
+    
+    originalCCMID = SHA-256(encode(crossChainMessageSchema, ccm))
+    return send(
+        EMPTY_FEE_ADDRESS,
+        ccm.moduleID,
+        ccm.crossChainCommandID,
+        ccm.sendingChainID,
+        0,
+        errorStatus,
+        ccm.params
+    )
+```
 
-##### Returns
+#### getChainAccount
 
-This function returns `chainAccount(chainID)`.
+The `getChainAccount` function returns the object stored in the chain data substore with store key equal to the input `chainID: ChainID`.
+It returns `chainAccount(chainID)`.
 
 #### getChannel
```

Comment:
You added the channel validators substore for which there is not getter at the moment.

Reply:
added